### PR TITLE
observability: add prometheus metrics

### PR DESCRIPTION
**Pull request by adelcast**
Add Prometheus metrics for Tornado server and Python Runtime. Other application specific metrics can be added in the future.

# Description

Prometheus style metrics are a vendor neutral standard based on a pull model. Prometheus-enabled servers output time-series data on a route (usually /metrics), which can be scraped. Being an open standard, most monitoring tools know how to interface with Prometheus metrics (Open Telemetry support them too).

More info on the Prometheus standard:

https://prometheus.io/docs/concepts/data_model/
https://github.com/prometheus/docs/blob/main/content/docs/instrumenting/exposition_formats.md

This PR enables the basic built-in metrics, which cover the Tornado server (http metrics) and the Python runtime. More metrics, specific to Mage can be added in the future

prometheus

# How Has This Been Tested?

Manually run 

./dev.sh default_repo
./dev.sh default_repo --enable_prometheus

Made sure /metrics is live only on the later case

- [ ] Test A
- [ ] Test B

# Checklist
- [ ] The PR is tagged with proper labels (bug, enhancement, feature, documentation)
- [ ] I have performed a self-review of my own code
- [ ] I have added unit tests that prove my fix is effective or that my feature works
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
  - [ ] If new documentation has been added, relative paths have been added to the appropriate section of `docs/mint.json`

cc:
@wangxiaoyou1993

### CODE REVIEW

**Comment by wangxiaoyou1993** [comment]
cc: @dy46

File: mage_ai/server/server.py [hunk 1]
```diff
@@ -309,6 +322,20 @@ def make_app(template_dir: str = None, update_routes: bool = False):
...
+        resource = Resource(attributes={
+            SERVICE_NAME: "mage"
+        })
+
+        # Initialize PrometheusMetricReader which pulls metrics from the SDK
+        # on-demand to respond to scrape requests
+        reader = PrometheusMetricReader()
+        provider = MeterProvider(resource=resource, metric_readers=[reader])
+        metrics.set_meter_provider(provider)
+        routes += [(r'/metrics', PrometheusMetricsHandler)]
```
[the remark this answers is not in the document]
Comment (adelcast) [reply mage_ai/server/server.py:337]: Its common practice to use /metrics, that is the [default](https://prometheus.io/docs/prometheus/latest/configuration/configuration/) that Prometheus uses to scrape.

[JupyterHub](https://jupyterhub.readthedocs.io/en/stable/reference/monitoring.html)
[Kafka](https://github.com/rama-nallamilli/kafka-prometheus-monitoring/blob/master/README.md)
[Postgres](https://github.com/prometheus-community/postgres_exporter) web.telemetry-path default is /metrics

I could use /metrics as a default, then allow config to overwrite it, but I think /metrics is pretty standard

[the remark this answers is not in the document]
Comment (adelcast) [reply mage_ai/server/server.py:337]: @dy46 let me know what you think....

[the remark this answers is not in the document]
Comment (dy46) [reply mage_ai/server/server.py:337]: ah ok i think i may have misunderstood what this endpoint is doing

File: mage_ai/server/server.py [hunk 2]
```diff
@@ -309,6 +322,20 @@ def make_app(template_dir: str = None, update_routes: bool = False):
         (r'/version-control', MainPageHandler),
     ]
 
+    if ENABLE_PROMETHEUS:
+        TornadoInstrumentor().instrument()
+        # Service name is required for most backends
+        resource = Resource(attributes={
+            SERVICE_NAME: "mage"
```
Comment (dy46) [inline mage_ai/server/server.py:329]: single quotes here too

Comment (adelcast) [reply mage_ai/server/server.py]: done

**Review by dy46 — APPROVED** [review]

**Pull request merged**